packages/next/src/cli/next-dev.ts

```diff
@@ -295,6 +295,12 @@ const nextDev: CliCommand = async (argv) => {
   } else {
     let cleanupFns: (() => Promise<void> | void)[] = []
     const runDevServer = async () => {
+      if (process.env.__INTERNAL_NEXT_DEV_TEST_TURBO_DEV) {
```

Comment:
This needs to fail higher up as it can be called via custom server and such directly bypassing the CLI

Reply:
afaik --turbo and any other turbopack related path does not work / considers custom server case even. does current test cases (using `launchapp`) have a setup that doesn't uses cli and need to pass --turbo / and custom server both?

Reply:
Yeah there are tests that bypass the CLI

Reply:
For those I'd like to approach with what Main mentioned - filter out tests that neither uses next-dev / turbopack and put it into permanent exclusion list for the turbopack tests.